### PR TITLE
Reduce default fee to 1000 zatoshis

### PR DESCRIPTION
**Pull request by daira**
Per ZIP 313. This also ensures that transactions that pay the default fee will always be relayed, and not rate-limited.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1603,8 +1603,8 @@ bool AcceptToMemoryPool(
         CTxMemPoolEntry entry(tx, nFees, GetTime(), dPriority, chainActive.Height(), mempool.HasNoInputsOf(tx), fSpendsCoinbase, consensusBranchId);
         unsigned int nSize = entry.GetTxSize();
 
-        // Accept a tx if it contains joinsplits and has at least the default fee specified by z_sendmany.
-        if (tx.vJoinSplit.size() > 0 && nFees >= ASYNC_RPC_OPERATION_DEFAULT_MINERS_FEE) {
+        // Accept a tx if it contains joinsplits and has at least the default fee for z_* operations.
+        if (tx.vJoinSplit.size() > 0 && nFees >= DEFAULT_FEE) {
```
Comment (nuttycom) [inline src/main.cpp:1607]: Should this be a constant like this, or could we plumb it all the way out to cli args?

Comment (daira) [reply src/main.cpp:1607]: Given that it can be overridden by the RPC caller already, I don't think it's necessary to have a config parameter.

**Review by rex4539 — CHANGES_REQUESTED** [review]
```
sprout_sapling_migration.py:
Initializing test directory /var/folders/pn/rswd5k6175d02hbt25tvl47r0000gn/T/test6eveymy2/545
Running test using '-migrationdestaddress'...
Mining blocks...
waiting for async operation opid-3327580f-3a9d-461f-96f8-2db044db4ea7
Mining to block 494 % 500...
migration operation: [{'id': 'opid-c72c37a4-53c7-4740-9f7e-fee0c1b56ecd', 'status': 'executing', 'creation_time': 1608285410, 'method': 'saplingmigration', 'target_height': 500}]
waiting for async operation opid-c72c37a4-53c7-4740-9f7e-fee0c1b56ecd
result: {'id': 'opid-c72c37a4-53c7-4740-9f7e-fee0c1b56ecd', 'status': 'success', 'creation_time': 1608285410, 'result': {'num_tx_created': 1, 'amount_migrated': '7.6999', 'migration_txids': ['137be0a0b3492e393c422b40478cebd67a638da96f190b1a0e4204711b8aec28']}, 'execution_secs': Decimal('39.325509'), 'method': 'saplingmigration', 'target_height': 500}
mempool: ['137be0a0b3492e393c422b40478cebd67a638da96f190b1a0e4204711b8aec28']
status: {'enabled': True, 'destination_address': 'zregtestsapling1ssqj3f3majnl270985gqcdqedd9t4nlttjqskccwevj2v20sc25deqspv3masufnwcdy67cydyy', 'unmigrated_amount': '2.30', 'unfinalized_migrated_amount': '7.6999', 'finalized_migrated_amount': '0.00', 'finalized_migration_transactions': 0, 'migration_txids': ['137be0a0b3492e393c422b40478cebd67a638da96f190b1a0e4204711b8aec28']}
Unexpected exception caught during testing: name 'DEFAULT_FEE' is not defined
Stopping nodes
Cleaning up
Failed

stderr:
  File "/Users/rex/rex4539-zcash-reduce-fee/qa/rpc-tests/test_framework/test_framework.py", line 143, in main
    self.run_test()
  File "/Users/rex/rex4539-zcash-reduce-fee/qa/rpc-tests/sprout_sapling_migration.py", line 184, in run_test
    self.run_migration_test(self.nodes[0], sproutAddr0, SAPLING_ADDR, 500)
  File "/Users/rex/rex4539-zcash-reduce-fee/qa/rpc-tests/sprout_sapling_migration.py", line 128, in run_migration_test
    assert_equal(Decimal('10.0') - DEFAULT_FEE, Decimal(status['unmigrated_amount']) + Decimal(status['unfinalized_migrated_amount']))

Pass: False, Duration: 240 s
```
```
mempool_limit.py:
Initializing test directory /var/folders/pn/rswd5k6175d02hbt25tvl47r0000gn/T/testuk7yn99m/526
Mining blocks...
Filling mempool...
Unexpected exception caught during testing: name 'DEFAULT_FEE' is not defined
Stopping nodes
Cleaning up
Failed

stderr:
  File "/Users/rex/rex4539-zcash-reduce-fee/qa/rpc-tests/test_framework/test_framework.py", line 143, in main
    self.run_test()
  File "/Users/rex/rex4539-zcash-reduce-fee/qa/rpc-tests/mempool_limit.py", line 64, in run_test
    opid1 = self.nodes[1].z_sendmany(get_coinbase_address(self.nodes[1]), [{"address": zaddr1, "amount": Decimal('10.0') - DEFAULT_FEE}])

Pass: False, Duration: 74 s
```
```
paymentdisclosure.py:
Initializing test directory /var/folders/pn/rswd5k6175d02hbt25tvl47r0000gn/T/test8v0jzb76/524
Mining blocks...
Unexpected exception caught during testing: name 'DEFAULT_FEE' is not defined
Stopping nodes
Cleaning up
Failed

stderr:
  File "/Users/rex/rex4539-zcash-reduce-fee/qa/rpc-tests/test_framework/test_framework.py", line 143, in main
    self.run_test()
  File "/Users/rex/rex4539-zcash-reduce-fee/qa/rpc-tests/paymentdisclosure.py", line 72, in run_test
    recipients = [{"address": myzaddr, "amount": Decimal('40.0') - DEFAULT_FEE}]

Pass: False, Duration: 70 s
```
```
signrawtransaction_offline.py:
Initializing test directory /var/folders/pn/rswd5k6175d02hbt25tvl47r0000gn/T/testo8e7gbmu/497
Mining blocks...
Unexpected exception caught during testing: name 'DEFAULT_FEE' is not defined
Stopping nodes
Cleaning up
Failed

stderr:
  File "/Users/rex/rex4539-zcash-reduce-fee/qa/rpc-tests/test_framework/test_framework.py", line 143, in main
    self.run_test()
  File "/Users/rex/rex4539-zcash-reduce-fee/qa/rpc-tests/signrawtransaction_offline.py", line 41, in run_test
    create_hex = self.nodes[0].createrawtransaction(create_inputs, {taddr: Decimal('10.0') - DEFAULT_FEE})

Pass: False, Duration: 42 s
```

**Comment by nuttycom** [comment]
@zkbot try

**Review by nuttycom — APPROVED** [review]
utACK

**Comment by nuttycom** [comment]
@zkbot r+

**Pull request merged**